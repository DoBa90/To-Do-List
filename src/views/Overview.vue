<template>
  <div class="overview">
    <div class="overview__headline">Ãœbersicht</div>
    <Tasks 
      class="overview__tasks"
      @deletedtasks="deleteSelectedTasks($event)"
      @activate-overlay="this.showOverlay = $event"
      :text="this.allTasks" 
    />
  </div>
    <Overlay 
      class="overview__overlay"
      v-if="showOverlay"
      @activate-overlay="this.showOverlay = $event"
      @click="openOverlay"
      @addtask="addNewTask($event)"
    />
</template>

<script>
import Overlay from '../components/Overlay.vue'
import Tasks from '../components/Tasks.vue'
import { getAllTasks, saveAllTasks, deleteTasks } from '../services/taskServices'

export default {
  name: 'App',
  components: {
    Overlay,
    Tasks
  },
  data() {
    return { 
      allTasks: [],
      showOverlay: false,
    }
  },
  mounted() {
      this.allTasks = getAllTasks()
  },
  computed: {
    savedTasks() {
      return this.tasks
    }
  },
  methods: {
    addNewTask(task) {
      this.allTasks.push(task)  
      saveAllTasks(this.allTasks)
    },

    deleteSelectedTasks(tasks) {
      console.log("deleteSelectedTasks", tasks)
      tasks.forEach(task => {
        const index = this.allTasks.indexOf(task)

        if(index !== -1) {
          deleteTasks(index, this.allTasks)
        }
      })
      saveAllTasks(this.allTasks);
    },
  },
}
</script>

<style lang="scss">

.overview {
    width: 100%;
    border-radius: 24px;
    background: #98a4da;
    margin: 12px;
    padding: 30px;

    &__headline {
        font-weight: 600;
        font-size: 21px;
        line-height: 36px;
        text-align: start;
        padding-bottom: 15px;
    }
}

@media only screen and (min-width: 768px) {
    .overview {
        width: 678px;
        min-height: 720px;
        padding: 60px;

        &__headline {
            font-size: 32px;
            line-height: 36px;
        }
    }
}
</style>
