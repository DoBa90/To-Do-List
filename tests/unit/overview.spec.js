import { shallowMount } from '@vue/test-utils'
import Overview from '@/views/Overview.vue'


const mountComp = options => shallowMount(Overview, {
  mocks: options?.mocks
});

describe('Overview.vue', () => {
  it('renders', () => {
    const overview = mountComp();

    expect(overview.exists()).toBeTruthy();
  // it('renders props.msg when passed', () => {
  //   const msg = 'new message'
  //   const wrapper = shallowMount(HelloWorld, {
  //     props: { msg }
  //   })
  //   expect(wrapper.text()).toMatch(msg)
  // })
  })

  it('renders variant "default" correctly', () => {
    const overview = mountComp();

    expect(overview.html()).toMatchSnapshot();
});
})



// import { createLocalVue, shallowMount } from '@vue/test-utils';
// import NavigationBox from '@/components/NavigationBox/NavigationBox.vue';

// const localVue = createLocalVue();

// const mountComp = options => shallowMount(NavigationBox, {
//     localVue,
//     propsData: {
//         headline: 'Hier alle GerÃ¤te ansehen und das richtige Produkt finden',
//         url: 'https://kaufland.de/elektronik',
//         href: 'https://kaufland.de/elektronik',
//         ...options?.propsData
//     },
//     mocks: options?.mocks
// });

// describe('NavigationBox', () => {
//     it('renders', () => {
//         const navigationBox = mountComp();

//         expect(navigationBox.exists()).toBeTruthy();
//     })