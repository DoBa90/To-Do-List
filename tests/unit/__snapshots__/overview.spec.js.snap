// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overview.vue renders variant "default" correctly 1`] = `
<div class="overview">
  <div class="overview__headline">Ãœbersicht</div>
  <tasks-stub text="" class="overview__tasks"></tasks-stub>
</div>
<!--v-if-->
`;
